crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -0,0 +1,696 @@
+                None => return Err(anyhow!("event without metadata")),
+            };
+            let ContractEvent::OrderPlacement(event) = data;
+            let order_data = convert_onchain_order_placement(
+                quoter,
+                &event,
+                quote_id,
+                domain_separator,
+                settlement_contract,
+            )
+            .await;
+            Ok((meta_to_event_index(&meta), order_data?))
+        },
+    );
+    let onchain_order_placement_data = futures::future::join_all(futures).await;
+    onchain_order_placement_data
+        .into_iter()
+        .filter_map(|data| match data {
+            Err(err) => {
+                tracing::debug!("Error while parsing onchain orders: {:}", err);
+                None
+            }
+            Ok((_, None)) => None,
+            Ok((event, Some(order_data))) => Some(Ok((event, order_data.0, order_data.1))),
+        })
+        .collect::<Result<Vec<_>>>()
+}
+
+async fn convert_onchain_order_placement(
+    quoter: &dyn OrderQuoting,
+    order_placement: &ContractOrderPlacement,
+    quote_id: i64,
+    domain_separator: DomainSeparator,
+    settlement_contract: H160,
+) -> Result<Option<(OnchainOrderPlacement, Order)>> {
+    let (order_data, owner, signing_scheme, order_uid) =
+        extract_order_data_from_onchain_order_placement_event(order_placement, domain_separator)?;
+
+    let quote_kind = convert_signing_scheme_into_quote_kind(signing_scheme, false)
+        .map_err(|err| anyhow!("Error invalid signature transformation: {:?}", err))?;
+
+    let parameters = QuoteSearchParameters {
+        sell_token: H160::from(order_data.sell_token.0),
+        buy_token: H160::from(order_data.buy_token.0),
+        sell_amount: order_data.sell_amount,
+        buy_amount: order_data.buy_amount,
+        fee_amount: order_data.fee_amount,
+        kind: order_data.kind,
+        // Original quote was made from user account, and not necessarily from owner.
+        from: order_placement.sender,
+        app_data: order_data.app_data,
+        quote_kind,
+    };
+    let quote = get_quote_and_check_fee(
+        quoter,
+        &parameters.clone(),
+        Some(quote_id as i64),
+        order_data.fee_amount,
+        model::quote::QuoteSigningScheme::Eip1271 {
+            onchain_order: true,
+        },
+    )
+    .await
+    .map_err(|err| {
+        anyhow!(
+            "Error while fetching the quote {:?} error: {:?}",
+            parameters,
+            err
+        )
+    })?;
+
+    let full_fee_amount = quote.data.fee_parameters.unsubsidized();
+
+    // Orders that are placed and priced outside the market (i.e. buying
+    // more than the market can pay or selling less than the market wants)
+    // get flagged as liquidity orders. The reasoning is that these orders
+    // are not intended to be filled immediately and so need to be treated
+    // slightly differently by the protocol.
+    let is_liquidity_order =
```

Comment:
It's not really fault of this PR, but I don't like to use the term "liquidity orders" to mean "orders that weren't created in the prescribed manner". It would be nicer to have a more clear term.